sdk/keyvault/azure-security-keyvault-jca/src/main/java/com/azure/security/keyvault/jca/JreCertificates.java

```diff
@@ -0,0 +1,167 @@
+// Copyright (c) Microsoft Corporation. All rights reserved.
+// Licensed under the MIT License.
+
+package com.azure.security.keyvault.jca;
+
+import java.io.IOException;
+import java.io.InputStream;
+import java.nio.file.Files;
+import java.nio.file.Path;
+import java.nio.file.Paths;
+import java.security.AccessController;
+import java.security.Key;
+import java.security.KeyStore;
+import java.security.KeyStoreException;
+import java.security.NoSuchAlgorithmException;
+import java.security.PrivilegedAction;
+import java.security.cert.Certificate;
+import java.security.cert.CertificateException;
+import java.util.stream.Stream;
+import java.util.Collections;
+import java.util.List;
+import java.util.Map;
+import java.util.Optional;
+import java.util.HashMap;
+import java.util.logging.Logger;
+
+import static java.util.logging.Level.WARNING;
+
+/**
+ * This class provides the certificates from jre key store. It only provides certificates. It does not provide key entries from jre key store.
+ */
+public final class JreCertificates implements AzureCertificates {
+    /**
+     * Stores the logger.
+     */
+    private static final Logger LOGGER = Logger.getLogger(JreCertificates.class.getName());
+
+    /**
+     * Stores the jre key store.
+     */
+    private final KeyStore jreKeyStore;
+
+    /**
+     * Stores the jre key store aliases.
+     */
+    private final List<String> aliases;
+
+    /**
+     * Stores the jre key store certificates.
+     */
+    private final  Map<String, Certificate> certs;
+
+    /**
+     * Stores the jre key store keys
+     */
+    private final  Map<String, Key> keys;
+
+    /**
+     * Stores the singleton
+     */
+    private static final JreCertificates INSTANCE = new JreCertificates();
+
+    /**
+     * Private constructor
+     */
+    private JreCertificates() {
+        jreKeyStore = JREKeyStore.getDefault();
+        aliases = Optional.ofNullable(jreKeyStore).map(a -> {
+            try {
+                return Collections.unmodifiableList(Collections.list(a.aliases()));
+            } catch (KeyStoreException e) {
+                LOGGER.log(WARNING, "Unable to load the jre key store aliases.", e);
+            }
+            return null; // It's ok to return null here. Null will be handled by Optional.
+        }).orElse(Collections.emptyList());
+        certs = aliases.stream()
+            .collect(HashMap::new, (m, v) -> {
+                try {
+                    m.put(v, jreKeyStore.getCertificate(v));
+                } catch (KeyStoreException e) {
+                    LOGGER.log(WARNING, "Unable to get the jre key store certificate.", e);
+                }
+            }, HashMap::putAll);
```

Comment:
nit:
```
        aliases = Optional.ofNullable(jreKeyStore)
                          .map(a -> {
                              try {
                                  return Collections.unmodifiableList(Collections.list(a.aliases()));
                              } catch (KeyStoreException e) {
                                  LOGGER.log(WARNING, "Unable to load the jre key store aliases.", e);
                              }
                              return null; // It's ok to return null here. Null will be handled by Optional.
                          })
                          .orElseGet(Collections::emptyList);
        certs = aliases.stream()
                       .collect(
                           HashMap::new,
                           (m, v) -> {
                               try {
                                   m.put(v, jreKeyStore.getCertificate(v));
                               } catch (KeyStoreException e) {
                                   LOGGER.log(WARNING, "Unable to get the jre key store certificate.", e);
                               }
                           },
                           HashMap::putAll);
```

1. Format code.
2. Use `orElseGet(Collections::emptyList)` instead of `orElse(Collections.emptyList())`